packages/api-headless-cms/src/crud/contentModel.crud.ts

```diff
@@ -183,30 +185,49 @@ export const createModelsCrud = (params: CreateModelsCrudParams): CmsModelContex
 
     const modelsList = async (): Promise<CmsModel[]> => {
         const databaseModels: CmsModel[] = await loaders.listModels.load("listModels");
-
         const pluginsModels = getModelsAsPlugins();
-
         return databaseModels.concat(pluginsModels);
     };
-
+    /**
+     * The list models cache is a key -> Promise pair so it the listModels() can be called multiple times but executed only once.
+     */
     const listModels = async () => {
-        return context.benchmark.measure("headlessCms.crud.models.listModels", async () => {
-            const models = await modelsList();
-            return filterAsync(models, async model => {
-                const ownsModel = await modelsPermissions.ensure(
-                    { owns: model.createdBy },
-                    { throw: false }
-                );
-
-                if (!ownsModel) {
-                    return false;
-                }
+        /**
+         * Maybe we can cache based on permissions, not the identity id?
+         *
+         * TODO: @adrian please check if possible.
+         */
+        const key = JSON.stringify({
```

Comment:
Do we really need this cache key thingie here?

I mean, for every HTTP request, there's really only one tenant / locale / identity combo right? If that's true, do we even need cache keys? Or we can just store the cache in a say `cache` variable directly and that's it?

Am I missing something here? 🤔

Reply:
Just ask @Pavel910 :)
Or you can think of how to fetch all models from a few diffrent tenant/locale in a single request without having this kind of cache key.